src/main/java/net/neoforged/neoforge/common/util/BlockSnapshot.java

```diff
@@ -120,18 +120,29 @@ public boolean restoreToLocation(LevelAccessor world, BlockPos pos, boolean forc
         if (world instanceof Level)
             ((Level) world).sendBlockUpdated(pos, current, replaced, flags);
 
+        restoreBlockEntity(world, pos);
+
+        if (DEBUG)
+            System.out.println("Restored " + this.toString());
```

Comment:
We should change this logic to use a slf4j logger. That way we can check `if (logger.isDebugEnabled())` or similar instead of using `System.out.println`.

Reply:
We can change it to emit to the main neo logger instead of println, but it still needs to be guarded behind the compile-time flag instead of checking `logger.isDebugEnabled()`

Reply:
That works - I was thinking of being in a world where we have the logging refactor work done :sweat_smile: